packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -299,41 +295,133 @@ class FlutterPreSubmitFileComparator extends FlutterGoldenFileComparator {
     final Platform platform, {
     SkiaGoldClient goldens,
     LocalFileComparator defaultComparator,
+    final Directory testBasedir,
   }) async {
 
     defaultComparator ??= goldenFileComparator as LocalFileComparator;
-    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(
+    final Directory baseDirectory = testBasedir ?? FlutterGoldenFileComparator.getBaseDirectory(
       defaultComparator,
       platform,
       suffix: '${math.Random().nextInt(10000)}',
     );
-    baseDirectory.createSync(recursive: true);
+
+    if (!baseDirectory.existsSync())
+      baseDirectory.createSync(recursive: true);
 
     goldens ??= SkiaGoldClient(baseDirectory);
-    await goldens.auth();
-    await goldens.tryjobInit();
-    return FlutterPreSubmitFileComparator(baseDirectory.uri, goldens);
+
+    final bool hasWritePermission = platform.environment['CIRRUS_USER_PERMISSION'] == 'admin'
+      || platform.environment['CIRRUS_USER_PERMISSION'] == 'write';
```

Comment:
Is this how we check elsewhere?

I think it would be better if we check the actual env variable that has the service account key and see if it starts with `ENCRYPTED[` or something like that.  That's what we really care about, right?

Reply:
Will do! :)